sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewTestSuite.scala

```diff
@@ -393,6 +393,18 @@ abstract class SQLViewTestSuite extends QueryTest with SQLTestUtils {
       assert(e2.message.contains("Cannot time travel views"))
     }
   }
+
+  test("SPARK-37569: view should report correct nullability information for nested fields") {
+    withView("v") {
+      val sql = "SELECT id, named_struct('a', id) AS nested FROM RANGE(10)"
+      val df = spark.sql(sql)
+
+      spark.sql(s"CREATE VIEW v AS $sql")
```

Comment:
We should call the `createView` method to create view, otherwise we just test the permanent view 3 times...

Reply:
Fixed it now